src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
+
+          // Clear socket data; we don't care what InstrumentApp sends
+          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
+          do
+            {
+              socket.Receive(buffer);
```

Comment:
This looks like it will block indefinitely:

From https://msdn.microsoft.com/en-us/library/8s4y8aff(v=vs.110).aspx
> If no data is available for reading, the Receive method will block until data is available, unless a time-out value was set by using Socket.ReceiveTimeout